### PR TITLE
[BugFix] Fix CLIPLoss targets shape in distributed training and missing return in all_gather. Co-authored with @qhua360

### PR DESCRIPTION
**Pull request by ahitagnied**
## Description

Fixes two bugs affecting distributed (DDP) training:

1. **`CLIPLoss` wrong targets under distributed training** (`losses/multimodal.py`):
   `_compute` all-gathers both sides, producing logits of shape `[world*B, world*B]`,
   but targets were always `arange(B)` instead of `arange(world*B)`. This caused a
   shape mismatch and incorrect loss values when running on multiple GPUs.

2. **`all_gather` silently dropped its return value** (`utils/distributed.py`):
   Missing `return` on the `torch.distributed.nn.functional.all_gather` call meant
   the function always returned the single-process fallback `(tensor,)` even in
   distributed mode.

**Validation:** Ran CLIP (RN50, bs512, lr0.002) under DDP before and after the fix,
and verified against a manual reference implementation. It was trained on CC3M over 40 epochs:
- `clip_rn50_20260408_132830` — pre-fix (buggy): plateaus at ~0.36 top-1
- `clip_rn50_20260410_014650` — post-fix: reaches ~0.42 top-1, matches manual impl
- `clip_rn50_20260409_132200` — manual reference impl ([OpenCLIP](https://github.com/mlfoundations/open_clip)): tracks post-fix run exactly: tracks post-fix run exactly

The pre-fix run shows clearly degraded performance, confirming the bug had a real
impact on training. Performance after the fix aligns with expected results from Table 2 of
[Sobal et al. (2024)](https://arxiv.org/abs/2407.18134).

<table>
  <tr>
    <td>linprobe/imagenet1k/top1</td>
    <td>linprobe/imagenet1k/top5</td>
  </tr>
</table>

## Checklist

- [x] I have read the [**Contributing**](https://rbalestr-lab.github.io/stable-SSL.github.io/dev/contributing.html#) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [ ] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR to the [**RELEASES.rst**](../RELEASES.rst) file.

### CODE REVIEW

**Review by RandallBalestriero — APPROVED** [review]

**Comment by RandallBalestriero** [comment]
@ahitagnied thanks for the PR! Ready to merge once you fix the last failing test.
